pkg/i2gw/ingress2gateway_test.go

```diff
@@ -158,3 +160,51 @@ func compareIngressLists(t *testing.T, gotIngressList *networkingv1.IngressList,
 		}
 	}
 }
+
+func Test_constructIngressesFromCluster(t *testing.T) {
+	ingress1 := networkingv1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "ingress1",
+			Namespace:       "namespace1",
+			ResourceVersion: "999",
+		},
+	}
+	ingress2 := networkingv1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "ingress2",
+			Namespace:       "namespace2",
+			ResourceVersion: "999",
+		},
+	}
+	testCases := []struct {
+		name        string
+		runtimeObjs []runtime.Object
+		wantIngress []networkingv1.Ingress
+	}{{
+		name:        "Test cluster client with 2 resources",
+		runtimeObjs: []runtime.Object{&ingress1, &ingress2},
+		wantIngress: []networkingv1.Ingress{ingress1, ingress2},
+	}, {
+		name:        "Test cluster client without resources",
+		runtimeObjs: []runtime.Object{},
+		wantIngress: []networkingv1.Ingress{},
+	},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ingressList := &networkingv1.IngressList{}
+			cl := fake.NewClientBuilder().WithRuntimeObjects(tc.runtimeObjs...).Build()
+			err := ConstructIngressesFromCluster(cl, ingressList)
+			if err != nil {
+				t.Errorf("test failed unexpectedly: %v", err)
+			}
+
+			got := []networkingv1.Ingress{}
+			got = append(got, ingressList.Items...)
+			if !apiequality.Semantic.DeepEqual(tc.wantIngress, got) {
```

Comment:
maybe apply ingresses1, 2 and and noNamespace that you already have above and use `compareIngressLists` function?